### PR TITLE
e2e,sgx: divide single It() to have multiple layers

### PR DESCRIPTION
**Pull request by hj-johannes-lee**
Structure is as follows:
```
Structure is as follows:
Describe("SGX plugin")
  BeforeEach("deploys plugin")
  Context("When device resources are available")
    BeforeEach("checks if resources are available")
    It("runs a pod requesting resources")
  AfterEach("undeploys plugin")
```
This is basically same structure as DSA, DLB, IAA, QAT E2E tests (#1263).

### CODE REVIEW

File: test/e2e/sgx/sgx.go [hunk 1]
```diff
@@ -48,104 +48,116 @@ func describe() {
...
+			dpPodName = podList.Items[0].Name
+
+			ginkgo.By("checking SGX plugin's securityContext")
+			if err = utils.TestPodsFileSystemInfo(podList.Items); err != nil {
+				framework.Failf("container filesystem info checks failed: %v", err)
+			}
+		})
+
+		ginkgo.AfterEach(func(ctx context.Context) {
+			ginkgo.By("undeploying DSA plugin")
```
Comment (mythi) [inline test/e2e/sgx/sgx.go:87]:
```suggestion
			ginkgo.By("undeploying SGX plugin")
```

File: test/e2e/sgx/sgx.go [hunk 2]
```diff
@@ -48,104 +48,116 @@ func describe() {
...
-		if err := utils.WaitForNodesWithResource(ctx, f.ClientSet, "sgx.intel.com/provision", 30*time.Second); err != nil {
-			framework.Failf("unable to wait for nodes to have positive allocatable provision resource: %v", err)
-		}
+			utils.Kubectl("", "delete", "-k", "deployments/operator/default/kustomization.yaml")
+		})
+	})
+}
 
-		utils.Kubectl("", "delete", "-f", "deployments/operator/samples/deviceplugin_v1_sgxdeviceplugin.yaml")
+func waitForNodesWithSgxResources(ctx context.Context, f *framework.Framework) {
```
Comment (mythi) [inline test/e2e/sgx/sgx.go:151]: we could think about can we check this in one place only for both operator and non-operator testing (now the operator rework PR moves the check to utils)

**Review by mythi — COMMENTED** [review]
let me know in which order you prefer to have the PRs handled

File: test/e2e/sgx/sgx.go (hunk 1 above)
Comment (hj-johannes-lee) [reply test/e2e/sgx/sgx.go]: Thanks, fixed!

**Comment by hj-johannes-lee** [comment]
@mythi When you have time, please take a look.! :)

**Review by mythi — APPROVED** [review]

**Pull request merged**